File: 232-implement-queue-using-stacks/implement-queue-using-stacks.cpp
class MyQueue {
public:
    stack<int> s1;
    MyQueue() {
        
    }
    
    void push(int x) {
        stack<int> s2;
       if(s1.empty()){
        s1.push(x);
        return;
        }
        while(!s1.empty()){
            s2.push(s1.top());
            s1.pop();
        }
        s1.push(x);
        while(!s2.empty()){
            s1.push(s2.top());
            s2.pop();
        }
    }
    
    int pop() {
       int r=s1.top();
       s1.pop();
        return r;
    }
    
    int peek() {
      return s1.top();  
    }
    
    bool empty() {
        if(s1.empty()) return true;
        return false;
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: 232-implement-queue-using-stacks/implement-queue-using-stacks.py
class MyQueue(object):

    def __init__(self):
        self.stack_in=[]

        

    def push(self, x):
        """
        :type x: int
        :rtype: None
        """
        self.stack_in.append(x)
        
        

    def pop(self):
        """
        :rtype: int
        """
        if self.empty():
            return -1
        a=self.stack_in[0]
        self.stack_in.pop(0)
        return a
        
        

    def peek(self):
        """
        :rtype: int
        """
        return self.stack_in[0]
        

    def empty(self):
        """
        :rtype: bool
        """
        if len(self.stack_in)==0:
            return True
        return False
        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()